### DISCUSSION TITLE
Cookie not stored correctly with Cypress

### DISCUSSION THREAD

**Discussion by abin53** [cypress-io/cypress#21509] [Questions and Help]
When I visit my site(https) directly, the certificate is verified, however the browser launched through Cypress is showing it as "Not Secure". Cypress says this is only cosmetic issue. But because of this our application fails since browser doesn't have a cookie with secure attribute set which in non-cypress browser works fine ? 

This doesn't happen while using chrome. But with other browsers like electron, pupputeer chrome etc

Any thoughts?  I'm new with Cypress. Point to me if I'm missing something.

**Comment by gplusdotgr** [comment]
@abin53 how do you set the cookie via Cypress? have you looked into https://docs.cypress.io/api/cypress-api/cookies ?

**Reply by abin53** [reply]
I dont set cookies via Cypress. Our application backend sets a cookie which is secure. And it works fine. Now since while I test/run our application within Cypress test runner the certificate is invalid it shows not secure therefore this cookie which I was talking about will not be set in Cypress browser. That's my concern.

![IMG_20220517_175420__01~2](https://user-images.githubusercontent.com/13628586/168810952-5da0b25a-3592-4aac-96bd-31055973150d.jpg)



Appreciate your reply.

**Reply by gplusdotgr** [reply]
You should mock the cookie and set it via Cypress following the guide above, don't count on your application to do this.

**Reply by abin53** [reply]
Thats not an option. This is csrf cookie which is validated at backend for every api calls(send as request header).

**Comment by abin53** [comment]
In continuation to above discussion 

I had to remove the 'secure' attribute from my cookie to continue

Still the cookie was not set in the browser for Cypress runs. I checked further to find out that:

Cypress make all set-cookie response headers into one single set-cookie header which makes cookies to be set with incorrect cookie path; which makes it an issue. 

Attaching some screenshots: 

1. Actual application in browser ( not with Cypress run )
![IMG_20220623_155525~3](https://user-images.githubusercontent.com/13628586/175285181-db16fa0f-a530-4733-b9ff-25ade93409af.jpg)

Note : multiple set cookie header present: XSRF-TOKEN is the cookie of discussion here which is set correctly with path = '/' and httpOnly as false

2. Application with Cypress run
![IMG_20220628_150148~2](https://user-images.githubusercontent.com/13628586/176146384-e9b7dd34-385c-47ce-92c4-50c7bf92a3dd.jpg)

Note : set-cookie header combined into single set-cookie. And also path '/scv' is incorrect. expected was '/'

3. Response cookie
![IMG_20220623_155722~3](https://user-images.githubusercontent.com/13628586/175281130-24bebbd5-25e0-43ab-9264-28d72d78508a.jpg)

The actual path for XSRF-TOKEN cookie was '/' but with single set-cookie and now it's being wrongly set as '/scv' which breaks the application under test. Also cookie is marked as httpOnly which isn't. 

Probably path and httonly are taken from the last cookies(JSESSIONID) information which is incorrect

Versions : 
Node 12.20.1
Cypress : 7.1.0

**Reply by abin53** [reply]
Checked for latest cypress version 10.3.0, still same issue is present

**Comment by alperen-bircak** [comment]
I am dealing with this issue as well, please revive this issue
